### PR TITLE
Update and clarify CI test script and usage instructions

### PR DESCRIPTION
**Pull request by saifshaikh48**

### CODE REVIEW

File: docs/HACKING.md [hunk 1]
```diff
@@ -273,10 +273,11 @@ Additional flags that can be passed to `hack/run-ci-e2e-test.sh` are
 - `-b` gives an alternative path to the WMCO binary. This option overridden in OpenShift CI.
        When building the operator locally, the WMCO binary is created as part of the operator image build process and
        can be found at `build/_output/bin/windows-machine-config-operator`, this is the default value of this option.
-- `-t` specify the test to run. All tests are run if the option is not used. The allowed options are:
-  - `all` all the tests are run
-  - `basic` creation, network and deletion tests are run
-  - `upgrade` creation, upgrade, reconfiguration and deletion tests are run
+- `-t` specify the test to run. All `maintanence` tests are run if the option is not used. The allowed options are:
+  - `maintanence` all the tests are run aside from the `upgrade-*` tests
```
Comment (jrvaldes) [inline docs/HACKING.md:277]: > `all` all the tests are run

I think, keeping  `all` makes more sense.

**Review by jrvaldes — APPROVED** [review]
LGTM

File: docs/HACKING.md (hunk 1 above)
Comment (saifshaikh48) [reply docs/HACKING.md:277]: happy to keep `all` if that's the consensus, though it is a bit of a misnomer since it doesn't run the upgrade tests

Comment (sebsoto) [reply docs/HACKING.md]: all is a misnomer.
Why not replace it with basic? Don't see why we need basic if its not being used in CI.

**Comment by saifshaikh48** [comment]
/unhold

azure upgrade job should be fixed now as https://github.com/openshift/release/pull/55411 merged

**Comment by saifshaikh48** [comment]
@jrvaldes Yep, thanks for the suggestion, in fact I would be in favor of overriding everything

Tests came back green and now tide kicked them ALL off again. This is making it excruciating to get any PRs in...

**Comment by mansikulkarni96** [comment]
/override ci/prow/aws-e2e-operator ci/prow/azure-e2e-operator ci/prow/azure-e2e-upgrade ci/prow/gcp-e2e-operator ci/prow/nutanix-e2e-operator ci/prow/platform-none-vsphere-e2e-operator ci/prow/vsphere-proxy-e2e-operator ci/prow/vsphere-e2e-operator ci/prow/wicd-unit-vsphere ci/prow/vsphere-disconnected-e2e-operator

**Pull request merged**